client_with_ssl.cpp

```diff
@@ -1,84 +1,224 @@
+{
+    static void write(const CityDTO& object, compact::compact_writer& out)
+    {
+        out.write_int32("population", object.population);
+        out.write_string("city", object.cityName);
+        out.write_string("country", object.country);
+    }
 
-void map_example(hazelcast::client::hazelcast_client client);
-void non_stop_map_example(hazelcast::client::hazelcast_client client);
+    static CityDTO read(compact::compact_reader& in)
+    {
+        CityDTO c;
 
-int main(int argc, char **argv) {
+        c.population = in.read_int32("population");
+        boost::optional<std::string> city = in.read_string("city");
+
+        if (city) {
+            c.cityName = *city;
+        }
+
+        boost::optional<std::string> country = in.read_string("country");
+
+        if (country) {
+            c.country = *country;
+        }
+
+        return c;
+    }
+
+    static std::string type_name() { return "city"; }
+};
+
+} // namespace serialization
+} // namespace client
+} // namespace hazelcast
+
+int
+main(int argc, char** argv)
+{
     hazelcast::client::client_config config;
+
     config.set_cluster_name("YOUR_CLUSTER_NAME");
-    config.set_property(hazelcast::client::client_properties::CLOUD_URL_BASE, "YOUR_DISCOVERY_URL");
+    config.set_property(hazelcast::client::client_properties::CLOUD_URL_BASE,
+                        "YOUR_DISCOVERY_URL");
     config.set_property("hazelcast.client.statistics.enabled", "true");
-    auto &cloud_configuration = config.get_network_config().get_cloud_config();
+
+    auto& cloud_configuration = config.get_network_config().get_cloud_config();
     cloud_configuration.enabled = true;
     cloud_configuration.discovery_token = "YOUR_CLUSTER_DISCOVERY_TOKEN";
+
     boost::asio::ssl::context ctx(boost::asio::ssl::context::tlsv12);
-    ctx.load_verify_file("ca.pem");
-    ctx.use_certificate_file("cert.pem", boost::asio::ssl::context::pem);
-    ctx.set_password_callback([&] (std::size_t max_length, boost::asio::ssl::context::password_purpose purpose) {
-        return "YOUR_SSL_PASSWORD";
-    });
-    ctx.use_private_key_file("key.pem", boost::asio::ssl::context::pem);
+
+    try {
+        ctx.load_verify_file("ca.pem");
+        ctx.use_certificate_file("cert.pem", boost::asio::ssl::context::pem);
+        ctx.set_password_callback(
+          [&](std::size_t max_length,
+              boost::asio::ssl::context::password_purpose purpose) {
+              return "YOUR_SSL_PASSWORD";
+          });
+        ctx.use_private_key_file("key.pem", boost::asio::ssl::context::pem);
+    } catch (std::exception& e) {
+        std::cerr << "You should copy ca.pem, cert.pem and key.pem files to "
+                     "the working directory, exception cause "
+                  << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    }
     config.get_network_config().get_ssl_config().set_context(std::move(ctx));
+    config.get_logger_config().level(hazelcast::logger::level::off);
```

Comment:
why?

Reply:
fixed at https://github.com/hazelcast/hazelcast-cloud-cpp-sample-client/pull/10/commits/a055c5e5b80c1df85ab07503c2ef208d66a3a643

Reply:
logs seems ugly when the example is run. So I made them off. Other clients also does not show logs.

Reply:
@ihsandemir all good by you? I can merge the changes after you resolve the review :)

Reply:
approved @EugeneAbramchuk